src/include/centering.hxx

```diff
@@ -0,0 +1,60 @@
+
+#pragma once
+
+#include "kg/Vec3.h"
+
+namespace Centering
+{
+
+enum CenterStyle
+{
+  NC, // node-centered
+  EC, // edge-centered
+  FC, // face-centered
+  CC, // cell-centered
+};
+
+enum Component
+{
+  X,
+  Y,
+  Z
+};
+
+/*  Centering Guide
+         X   Y   Z
+       -------------
+NC  |   nnn nnn nnn
+EC  |   cnn ncn nnc
+FC  |   ncc cnc ccn
+CC  |   ccc ccc ccc
+*/
+
+template <typename PATCH>
+Double3 getPos(PATCH patch, Int3 index, int style, int comp = X)
```

Comment:
There may be some point in making `style` to be a template arg at some point, so that the different cases can be optimized at compile time. (Or maybe that won't work for all use cases. In any case, it's fine for the time being.)